src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -72,7 +72,7 @@ private set
                 {
                     lock (_statusLock)
                     {
-                        if (_status != value)
+                        if (_status != value && _status != ServiceConnectionStatus.Terminated)
```

Comment:
why adding a status? a `_close` bool is enough?

Reply:
This is better? Terminated means the connection cannot reconnect anymore. It let others know the current connection's status

Reply:
No. I don't think ServerConnection itself is aware of "reconnect" stuff. Introducing in a new concept makes it more complex. The container is where to determine how it handles the connections.